client/components/ReviewRating/WriteReview/WriteReview.jsx

```diff
@@ -0,0 +1,195 @@
+import React, { useState, useEffect } from 'react';
+
+import PropTypes from 'prop-types';
+import axios from 'axios';
+
+const WriteReview = ({ isClickAdd, closeWriteReview, id }) => {
+  const [newReview, setNewReview] = useState({
+    rating: 5,
+    summary: '',
+    body: '',
+    recommend: true,
+    name: '',
+    email: '',
+    photos: '',
+    characteristics: {},
+    product_id: id,
+  });
+  const [isValid, setIsValid] = useState(true);
+  const [photoUrl, setPhotoUrl] = useState([]);
+  const [charName, setCharName] = useState([]);
+  const [charId, setCharId] = useState([]);
+  const [charValue, setCharValue] = useState([]);
+
+  useEffect(() => {
+    axios.get(`/reviews/meta/${id}`)
+      .then((res) => {
+        setCharName(Object.entries(res.data.characteristics).map((i) => i[0]));
+        setCharId(Object.entries(res.data.characteristics).map((i) => i[1]).map((i) => i.id));
+        // eslint-disable-next-line max-len
+        setCharValue(Object.entries(res.data.characteristics).map((i) => i[1]).map((i) => parseInt(i.value, 10)));
+      });
+  }, []);
+
+  const looksLikeMail = (str) => {
```

Comment:
Looks like a utility function to me. Maybe consider importing from our utils file.

Reply:
That is the function to validate the email input. I don't think other components can apply to it. But if the utility will be a good place to go. I can move it.

Reply:
It's up to you. Just asking questions for my own clarification. Thanks.